sources/Application/Instruments/SamplePool.cpp

```diff
@@ -166,7 +166,12 @@ bool SamplePool::loadSample(const char *path) {
   WavFile *wave = WavFile::Open(path);
   if (wave) {
     wav_[count_] = wave;
-    const std::string name = wavPath.GetName();
+    std::string name = wavPath.GetName();
+    // Drop file extension
+    size_t index = name.find_last_of(".");
+    if (index < name.length() && index > 0) {
```

Comment:
This seems unnecessary, by definition of the find_last_of function, the index will necessary be within bounds of the string. One thing that could happen is that the '.' is not found, in which case you'd get `string::npos` and thus...